site-src/geps/gep-1282.md

```diff
@@ -98,12 +98,302 @@ Of course, from the section above, you can see that there are many other feature
+   // characteristics in common: they are tightly bound to the backend, 
+   // rather than being important at the route level, and the service owner // (application developer in the Gateway API personas) should control 
+   // these settings, not the owner of the Gateway implementation (the 
+   // Cluster Operator in the Gateway API personas).
+   // 
+   // There are a maximum of 32 BackendObjectCapabilities per BackendRef.
+   //
+   // Support: Extended
+   //
+   // +optional
+   // +kubebuilder:validation:MaxItems=32
+   BackendObjectCapabilities *BackendObjectCapabilitiesMap `json:”backendObjectCapabilitiesMap,omitempty”`
+	
+	// Weight specifies the proportion of requests forwarded to the referenced
+	//...
+	Weight *int32 `json:"weight,omitempty"`
+}
+
+// BackendObjectCapabilitiesMap defines a mapping of BackendObjectCapability name
+// to an array of BackendObjectCapability
+BackendObjectCapabilitiesMap map[string][]BackendObjectCapability
+
+// BackendObjectCapability defines a capability or property held by its
+// affiliated BackendObject, and specified by the BackendObject owner 
+// (typically the application developer).  Examples of BackendObjectCapability
+// use cases include: TLS re-encryption, session state, timeouts, retries, 
+// Websockets, HTTP/2, load balancer algorithm, healthcheck info, external
+// traffic policy, connection limits, error page, protocol switch, alerts. 
+//
+// The BackendObject must be valid in the cluster; the Group and Kind must
+// be registered in the cluster for this reference to be valid.
+//
+// Support: Extended
+//
+// References to objects with invalid Group and Kind are not valid.
+type BackendObjectCapability struct {
+	// Group is the group of the referent.
+	//
+	// Support: Core
+	//
+	// +kubebuilder:default=gateway.networking.k8s.io
+	// +optional
+	Group *Group `json:"group,omitempty"`
+
+	// Kind is kind of the referent.
+	//
+	// Support: Custom
+	//
+	// +kubebuilder:default=Gateway
+	// +optional
+	Kind *Kind `json:"kind,omitempty"`
+
+	// Name is the name of the referent.
+	//
+	// Support: Core
+	Name ObjectName `json:"name"`
+
+	// Value is the value of the referent.  The type is not specifically
+	// defined and is left up to the implementation to infer.
+	//
+	// Support: Extended
+	// 
+	// +optional
+	Value interface{} `json:”value”`
+}
+
+```
+### Example Capabilities for Alternative 1
+
+TLS information for connection from the Gateway to the backend service. What a Gateway’s proxy would need to be able to
+connect to the backend service.
+
+The BackendCapabilities could be defined by an implementation and look something like:
+
+```yaml
+
+backendCapabilities:
+  tls-reencrypt:
+    - name: version
+    value: 1.3
+    - name: clientCert
+    value: /path/to/cert.pem
+    - name: privateKey
+    value: /path/to/pkey.pem
+    - name: caCerts
+    value: /path/to/cacerts.pem
+    - name: backendCert
+    value: /path/to/backendCert.pem
```

Comment:
Likely a strange question: what is the syntax used in the examples in this chapter? :-) I was rather expecting to see (partial) Kubernetes manifests matching the API / CRD proposal earlier in this document instead of general name/value records.

Reply:
This is an Alternative option that was considered, a map of structs, offering the most minimal set of predefined fields.  We will not likely choose this, but if anyone has a preference for `Alternative 1`, now would be the time to discuss that.